source/common/router/config_impl.cc

```diff
@@ -45,6 +45,11 @@ namespace Envoy {
 namespace Router {
 namespace {
 
+absl::string_view getXFPOrScheme(const Http::RequestHeaderMap& headers) {
+  absl::string_view xfp = headers.getXForwardedProtoValue();
+  return xfp.empty() ? headers.getSchemeValue() : xfp;
+}
```

Comment:
Not sure if this is worth co-locating with Utility::getScheme for comment clarity? At first glance I thought it was the same vs. the inverse.